libs/perun/components/src/lib/members-list/members-list.component.ts

```diff
@@ -83,12 +84,45 @@ export class MembersListComponent implements OnChanges, AfterViewInit {
   dataSource: MatTableDataSource<RichMember>;
   pageSizeOptions = TABLE_ITEMS_COUNT_OPTIONS;
 
+  getSortDataForColumn(data: RichMember, column: string, outerThis: MembersListComponent): string{
+    switch (column) {
+      case 'id':
+        return data.id.toString();
+      case 'fullName':
+        return parseFullName(data.user);
+      case 'status':
+        return  outerThis.showGroupStatuses ? data.groupStatus : data.status;
+      case 'organization':
+        return parseOrganization(data);
+      case 'email':
+        return parseEmail(data);
+      default:
+        return '';
+    }
+  }
+
+  getFilterDataForColumn(data: RichMember, column: string): string{
+    switch (column) {
+      case 'fullName':
+        return parseFullName(data.user);
+      case 'email':
+        return parseEmail(data);
+      case 'logins':
+        return parseLogins(data);
+      default:
+        return '';
+    }
+  }
+
   setDataSource() {
     this.displayedColumns = this.displayedColumns.filter(x => !this.hideColumns.includes(x));
     if (!!this.dataSource) {
-      this.dataSource.filterPredicate =
-        (data: RichMember, filter: string) => parseFullName(data.user).toLowerCase().includes(filter.toLowerCase()) ||
-          data.id.toString(10).includes(filter);
+      this.dataSource.filterPredicate = (data: RichMember, filter: string) => {
+        return customDataSourceFilterPredicate(data, filter, this.displayedColumns, this.getFilterDataForColumn, this);
+      };
+      this.dataSource.sortData = (data: RichMember[], sort: MatSort) => {
+        return customDataSourceSort(data, sort, this.getSortDataForColumn, this);
+      };
```

Comment:
![image](https://user-images.githubusercontent.com/32871167/107011027-65c40580-6797-11eb-989f-b252ea99044a.png)
this is the desired sorting in member-list when it comes to sorting according to names
primarily we want to sort according to what is in the "lastName" if last name is null then sort according to "firstName" if first name is null then sort according to something else so the method at least doesnt throw error

Reply:
![image](https://user-images.githubusercontent.com/32871167/107011294-bdfb0780-6797-11eb-839a-ca777c261daf.png)
this is your sorting according to full name, you are even counting the titles..

Reply:
same problem is everywhere where is fullName of user sorted - application-list, candidate-list